modules/credential/src/main/java/org/eclipse/xpanse/modules/credential/CredentialCenter.java

```diff
@@ -174,7 +179,15 @@ public void addCredential(CreateCredential createCredential) {
                     String.format("Not supported credential type Csp:%s, Type: %s.",
                             createCredential.getCsp(), createCredential.getType()));
         }
-        createCredential(credential);
+    }
+
+    private void encodeSensitiveVariables(CreateCredential createCredential) {
+        List<CredentialVariable> variables = createCredential.getVariables();
+        variables.stream().forEach(variable -> {
+            if (!Objects.isNull(variable) && variable.getIsSensitive()) {
+                variable.setValue(passwordEncoder.encode(variable.getValue()));
```

Comment:
When variable.getValue() is null, encode (null) will throw NullPointerException，We should add a judgment condition Objects.isNull(variable.getValue()) ;